src/main/java/cloud/fogbow/ras/core/plugins/interoperability/aws/compute/v2/AwsV2ComputePlugin.java

```diff
@@ -1,47 +1,459 @@
+		String name = defineInstanceName(computeOrder.getName());
+
+		Tag tagName = Tag.builder()
+				.key(AWS_TAG_NAME)
+				.value(name)
+				.build();
+
+		CreateTagsRequest tagRequest = CreateTagsRequest.builder()
+				.resources(instanceId)
+				.tags(tagName)
+				.build();
+
+		return tagRequest;
+	}
+
+	protected String defineInstanceName(String instanceName) {
+		return instanceName == null ? SystemConstants.FOGBOW_INSTANCE_NAME_PREFIX + getRandomUUID() : instanceName;
+	}
+
+	// This method is used to aid in the tests
+	protected String getRandomUUID() {
+		return UUID.randomUUID().toString();
+	}
+
+	protected AwsHardwareRequirements findSmallestFlavor(ComputeOrder computeOrder, AwsV2User cloudUser)
+			throws InvalidParameterException, UnexpectedException, ConfigurationErrorException,
+			NoAvailableResourcesException {
+
+		AwsHardwareRequirements bestFlavor = getBestFlavor(computeOrder, cloudUser);
+		if (bestFlavor == null) {
+			throw new NoAvailableResourcesException(Messages.Exception.NO_MATCHING_FLAVOR);
+		}
+		return bestFlavor;
+	}
+
+	protected AwsHardwareRequirements getBestFlavor(ComputeOrder computeOrder, AwsV2User cloudUser)
+			throws InvalidParameterException, UnexpectedException, ConfigurationErrorException {
+
+		updateHardwareRequirements(cloudUser);
+		TreeSet<AwsHardwareRequirements> resultset = getFlavorsByRequirements(computeOrder.getRequirements());
+		for (AwsHardwareRequirements hardwareRequirements : resultset) {
+			if (hardwareRequirements.getCpu() >= computeOrder.getvCPU()
+					&& hardwareRequirements.getMemory() >= computeOrder.getMemory()
+					&& hardwareRequirements.getDisk() >= computeOrder.getDisk()) {
+				return hardwareRequirements;
+			}
+		}
+		return null;
+	}
+
+	protected TreeSet<AwsHardwareRequirements> getFlavorsByRequirements(Map<String, String> orderRequirements) {
+		TreeSet<AwsHardwareRequirements> resultSet = getFlavors();
+		List<AwsHardwareRequirements> resultList = null;
+		if (orderRequirements != null && !orderRequirements.isEmpty()) {
+			for (Entry<String, String> requirements : orderRequirements.entrySet()) {
+				resultList = filterFlavors(requirements);
+				if (resultList.size() < resultSet.size()) {
+					resultSet = parseToTreeSet(resultList);
+				}
+			}
+		}
+		return resultSet;
+	}
+	
+	protected TreeSet<AwsHardwareRequirements> parseToTreeSet(List<AwsHardwareRequirements> list) {
+		TreeSet<AwsHardwareRequirements> resultSet = new TreeSet<AwsHardwareRequirements>();
+		resultSet.addAll(list);
+		return resultSet;
+	}
+
+	protected List<AwsHardwareRequirements> filterFlavors(Entry<String, String> requirements) {
+		String key = requirements.getKey().trim();
+		String value = requirements.getValue().trim();
+		return getFlavors().stream()
+				.filter(flavor -> flavor.getRequirements().get(key).equalsIgnoreCase(value))
+				.collect(Collectors.toList());
+	}
+	
+	protected TreeSet<AwsHardwareRequirements> getFlavors() {
+		synchronized (this.flavors) {
```

Comment:
why is it needed?

Reply:
This model is being replicated from other plugin implementations. I believe it is to avoid competition for the use of the feature when someone is accessing the stack, another may be trying to update it.